### DISCUSSION TITLE
Axios doesn't work in APK

### DISCUSSION THREAD

**Discussion by developerSave** [quasarframework/quasar#14966] [CLI - Capacitor / Cordova modes] [answered]
Hi everyone,
I've a problem with my axios calls in my android app based on Quasar-Capacitor-Vue2.
In my frontend code, for example, the axios calls the login method "https://my-site.com/api/public/login". In the quasar.conf.js file I've the same host as the axios login and in the backend's method I've a link to the same address.
When I start the "quasar build" command I create the files that I'll use in my aws server, which points to the address "https://my-site-2.com/api/public/login" and I don't have any problems logging in from browser.
After that, when I generate the apk from android studio (build -> generate signed apk -> apk), I get the login page, but the axios doesn't work.

What am I doing wrong?

Regards,
Lorenzo

**Comment by gaguerrero** [comment, marked as answer]
In case it helps you, I used the cordova-plugin-advanced-http.

**Reply by developerSave** [reply]
Hi!
Can I use capacitor and cordova together? And I also have another problem:
I put my url ("https://my-site.com/api/") in:

quasar.conf.js  file
    capacitor: {
      hideSplashscreen: true,
      "server":{
        "hostname": "https://my-site.com/api/",
        "androidScheme": "https"
        }
    },

    devServer: {
      https: true,
      port: 8443,
      host: 'https://my-site.com/api/'
}


capacitor.config.json  both files
    "server": {
    "hostname": "https://my-site.com/api/",
    "androidScheme": "https"
  }

but when I build my apk from Android studio, it gets localhost or http://https//my-site.com/api/ .
How can I fix that?
Thank you!

**Reply by metalsadman** [reply]
some cordova plugins works in capacitor, or try capacitor's own plugin found in their docs here https://capacitorjs.com/docs/apis/http

**Reply by developerSave** [reply]
Thank you!

**Comment by gaguerrero** [comment]
I apologize, I meant that perhaps it could help you to evaluate the use of cordova. I have no experience in Capacitor.

**Comment by TobyMosque** [comment]
read that thread:
https://forum.ionicframework.com/t/axios-http-request-not-working-on-a-physical-device/194813/10

**Comment by metalsadman** [comment]
been using axios in cordova or capacitor and working fine on dev and or build for me whether http/https provided my server was set for cors.

**Comment by developerSave** [comment]
thank you, guys!
Now I'm going to fix other problems in my code
